### DISCUSSION TITLE
CDK Context - Required and validation (Typescript)

### DISCUSSION THREAD

**Discussion by falnyr** [aws/aws-cdk#15393] [Q&A]
Hi folks,
I'm trying to understand whether it is possible to validate [context](https://docs.aws.amazon.com/cdk/latest/guide/context.html) values.

Essentially, I am looking for a use case like:

```typescript
// my-stack.ts

interface MyStackContext {
    requiredAttribute: string;
}

export class MyStack extends cdk.Stack {
    constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props);

        // validating the attributes
        validateAttributes(scope.node.tryGetContext("requiredAttribute"))
    }
}
```

Is there a native support for this or do I have to write my custom logic?

Thanks!

**Comment by peterwoodworth** [comment]
I'm not sure what we'd be validating exactly, this sounds like something you'll need to write for your own use case
